architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,276 @@
+
+**Mitigations:**
+
+1. Secret data managed by the UCP is stored at rest in Kubernetes secrets or etcd. Our threat model assumes that the API server and mechanisms like Kubernetes authentication/RBAC are not compromised.
+2. Secrets managed by Radius are always placed in the same namespace as the object that "owns" them. This is a requirement of the Kubernetes RBAC model.
+3. Secrets managed by Radius are subject to the Kubernetes RBAC model for controlling access. Operators are expected to limit access for users using existing tools.
+
+**Status:** All mitigations listed are currently active. Operators are expected to secure their cluster and limit access for users.
+
+#### Threat: Escalation of Privilege by Using Radius to Circumvent Kubernetes RBAC Controls
+
+**Description:** A malicious actor could circumvent Kubernetes RBAC controls and create arbitrary resources in Kubernetes by using Universal Control Plane. UCP operates with a wide scope of permissions in Kubernetes and the cloud.
+
+At the time of writing, Radius does not provide granular authorization controls. Any authenticated client can create any Radius resource and execute any action Radius is capable of taking. This is the case in UCP as well as other components of Radius like the Kubernetes controllers.
+
+**Impact:** An authorized user of the Kubernetes cluster with permission to create a resource can execute any application definition in any environment registered with Radius.
+
+**Mitigations:**
+
+1. Operators should limit direct access to the Radius API using Kubernetes RBAC.
+2. We should revisit the threat model and provide a more robust set of authorization controls when granular authorization policies are added to Radius.
+
+**Status:** These mitigations are partial and require configuration by the operator. We will revisit and improve this area in the future.
+
+#### Threat: Lack of Role-Based Access Control (RBAC) and Unauthorized Traffic
+
+**Description:** As mentioned above, as of November 15, 2024, the UCP does not implement RBAC, and communication between the UCP and resource providers is unauthenticated.
+
+**Impact:** Increased risk of unauthorized access and actions, making it easier for attackers to interact with resource providers or manipulate user resources without proper authorization.
+
+**Mitigations:**
+
+1. **Implement RBAC within the UCP:** An authentication and authorization mechanism that verifies the identity of clients and enforces access policies must be developed.
+2. **Secure Communication Between UCP and Other Components:** mTLS should be enabled.
+3. **Network Policies and Firewall Rules:** Application of Kubernetes Network Policies to control traffic flow to and from the UCP.
+
+**Status:** Other than the second mitigation (secure communication) listed above, none of the mitigations is currently active.
+
+#### Threat: Use of Weak Cryptographic Hashing Algorithm (SHA-1)
```

Comment:
Can you be more specific about the threat? Complete the following sentence:

> A malicious actor could use X to do Y...

This reads like an action-item not a threat.